docs/proposals/004-cluster-name-as-id.md

```diff
@@ -0,0 +1,149 @@
+---
+title: Cluster Name as ID
+authors:
+  - "@denysvitali" # Authors' github accounts here.
+sponsors:
+  - TBD        # List all interested parties here.
+reviewers:
+  - "@alexmt"
+  - TBD
+approvers:
+  - "@alexmt"
+  - TBD
+
+creation-date: 2023-03-07
+last-updated: 2023-03-07
+---
+
+# Cluster Name as unique identifier
+
+
+<!-- ## Open Questions [optional] -->
+
+
+## Summary
+
+Currently, the clusters are identified by their API URL. Pretty much everywhere in the codebase, we assume that the API
+URL is unique within an ArgoCD instance, and thus this is used as an identifier.  
+With this proposal, we suggest to switch to a (Namespaced) Cluster Name oriented approach, so that one can use the same API URL multiple times
```

Comment:
When we say Cluster Name, does it mean the name of the cluster specified in the `.data.name` of the secret? Or is it the name of the secret resource `metadata.name`?  Since `.data.name` is not a mandatory field it could be difficult to check for the uniqueness of the key. However, the secret name is mandatory, and combining it with the namespace will result in a unique key.